providers/oidc.go

```diff
@@ -168,11 +175,27 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	newSession.AccessToken = token.AccessToken
 	newSession.RefreshToken = token.RefreshToken
 	newSession.CreatedAt = &created
-	newSession.ExpiresOn = &token.Expiry
```

Comment:
I think we can just leave this logic as is and do all this math in the centralized spot based on the `CreatedAt` and `ExpiresOn` we set here.

We'll just refresh before the expiration at the hardcoded time or the at the % of the lifespan.

Reply:
I don't know if we can. Providers can have different behaviors, use different fields for the same computations etc. I'm not dead set against it, just don't want generalize too much. Plus I'm personally only interested in generic OIDC and maybe Keycloak as a provider, so I can't speak about the behaviors of others and their quirks.

Reply:
Let's have all the timing go off our session timings and lifespans (it is a `cookie` refresh after all) -- simplifies the logic between all the providers.  And skip the provider introspection into all the potential access/ID Token timing claims.

I think users who care about the sessions corresponding to the underlying tokens (if they care to pass tokens to the upstream) are already aware of syncing the lifespans of the two in the configurations.

Reply:
For your case with OIDC -- if we already match the `session.CreatedAt` to the creation and `session.ExpiresOn` to the `token.Expiry` (like we do now) -- you still get your percentage TTL behavior based on the token lifespans  that you want by implementing in the common/global area, right?